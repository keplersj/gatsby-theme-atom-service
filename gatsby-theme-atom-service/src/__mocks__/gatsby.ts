import * as React from "react";
const gatsby = jest.requireActual("gatsby");

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest.fn().mockImplementation(
    ({
      // these props are invalid for an `a` tag
      /* eslint-disable @typescript-eslint/no-unused-vars */
      /* eslint-disable no-unused-vars */
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      /* eslint-enable @typescript-eslint/no-unused-vars */
      /* eslint-enable no-unused-vars */
      to,
      ...rest
    }): React.ReactElement =>
      React.createElement("a", {
        ...rest,
        href: to
      })
  ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn()
};
