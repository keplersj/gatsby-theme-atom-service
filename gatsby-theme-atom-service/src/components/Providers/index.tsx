import { graphql, useStaticQuery } from "gatsby";
import { sortBy } from "lodash";
import * as React from "react";
import { Grid } from "@material-ui/core";
import { JsonLd } from "react-schemaorg";
import { SoftwareApplication } from "schema-dts";
import { Card } from "../Card";

interface ProvidersQuery {
  providers: {
    edges: {
      node: {
        id: string;
        title: string;
        modal: string;
        types: {
          title: string;
          modal: string;
          packages: {
            title: string;
            url: string;
          }[];
        }[];
      };
    }[];
  };
}

const Providers = (): React.ReactElement => {
  const data = useStaticQuery<ProvidersQuery>(graphql`
    query ProvidersQuery {
      providers: allProvidersYaml {
        edges {
          node {
            id
            title
            modal
            types {
              title
              modal
              packages {
                title
                url
              }
            }
          }
        }
      }
    }
  `);

  return (
    <>
      {data.providers.edges.map(({ node: scope }) => (
        <div key={scope.id} id={scope.modal}>
          <h2>{scope.title}</h2>
          <Grid container justify="space-evenly">
            {sortBy(scope.types, "title").map((type) => (
              <React.Fragment key={type.title}>
                {type.packages.map((package_) => (
                  <JsonLd<SoftwareApplication>
                    key={package_.url}
                    item={{
                      "@context": "https://schema.org",
                      "@type": "SoftwareApplication",
                      "@id": package_.url,
                      applicationCategory: "Development Tool",
                      applicationSubCategory: "Atom Package",
                      name: package_.title,
                      url: package_.url,
                      mainEntityOfPage: package_.url,
                      operatingSystem: "macOS, Windows, Linux",
                    }}
                  />
                ))}
                <Card
                  title={type.title}
                  links={type.packages.map((package_) => ({
                    name: package_.title,
                    url: package_.url,
                  }))}
                />
              </React.Fragment>
            ))}
          </Grid>
        </div>
      ))}
    </>
  );
};

export default Providers;
