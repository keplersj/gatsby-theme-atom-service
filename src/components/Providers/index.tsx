import { graphql, useStaticQuery } from "gatsby";
import { sortBy } from "lodash";
import * as React from "react";
import { Grid } from "@material-ui/core";
import { Card } from "../Card";

interface ProvidersQuery {
  providers: {
    edges: {
      node: {
        id: string;
        title: string;
        plural: string;
        modal: string;
        types: {
          title: string;
          modal: string;
          packages: {
            title: string;
            url: string;
          }[];
        }[];
      };
    }[];
  };
}

const Providers = () => {
  const data = useStaticQuery<ProvidersQuery>(graphql`
    query ProvidersQuery {
      providers: allProvidersYaml {
        edges {
          node {
            id
            title
            plural
            modal
            types {
              title
              modal
              packages {
                title
                url
              }
            }
          }
        }
      }
    }
  `);

  return (
    <>
      {data.providers.edges.map(({ node: scope }) => (
        <div key={scope.id} id={scope.modal}>
          <h2>{scope.title}</h2>
          <Grid container justify="space-evenly">
            {sortBy(scope.types, "title").map(type => (
              <Card
                title={type.title}
                links={type.packages.map(pkg => ({
                  name: pkg.title,
                  url: pkg.url
                }))}
              />
            ))}
          </Grid>
        </div>
      ))}
    </>
  );
};

export default Providers;
