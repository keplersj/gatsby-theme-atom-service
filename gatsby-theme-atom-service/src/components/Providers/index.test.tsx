import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import Providers from ".";

describe("Providers", () => {
  beforeEach(() => {
    (useStaticQuery as jest.Mock).mockImplementationOnce(() => ({
      providers: {
        edges: [
          {
            node: {
              id: "123",
              title: "Example Category",
              modal: "category",
              types: [
                {
                  title: "Example Type",
                  modal: "example",
                  packages: [
                    {
                      title: "Example Package",
                      url: "https://example.dev/provider"
                    }
                  ]
                }
              ]
            }
          }
        ]
      }
    }));
  });

  it("renders as expected", () => {
    const tree = renderer.create(<Providers />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
