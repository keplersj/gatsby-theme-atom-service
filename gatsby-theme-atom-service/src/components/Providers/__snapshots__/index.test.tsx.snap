// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Providers renders as expected 1`] = `
<div
  id="category"
>
  <h2>
    Example Category
  </h2>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-evenly"
  >
    <script
      type="application/ld+json"
    >
      Object {
      "@context": "https://schema.org",
      "@id": "https://example.dev/provider",
      "@type": "SoftwareApplication",
      "applicationCategory": "Development Tool",
      "applicationSubCategory": "Atom Package",
      "mainEntityOfPage": "https://example.dev/provider",
      "name": "Example Package",
      "operatingSystem": "macOS, Windows, Linux",
      "url": "https://example.dev/provider",
    }
    </script>
    <div
      className="MuiGrid-root makeStyles-gridItem-104 MuiGrid-item"
    >
      <div
        className="MuiPaper-root MuiCard-root makeStyles-cardBase-105 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiCardContent-root"
        >
          <h2>
            Example Type
          </h2>
        </div>
        <hr
          className="MuiDivider-root"
        />
        <div
          className="MuiCardActions-root MuiCardActions-spacing"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap"
          >
            <a
              aria-disabled={false}
              className="MuiButtonBase-root MuiButton-root MuiButton-text"
              href="https://example.dev/provider"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
            >
              <span
                className="MuiButton-label"
              >
                Example Package
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
